zcash_client_sqlite/src/wallet.rs

```diff
@@ -157,29 +163,90 @@ pub(crate) fn add_account<P: consensus::Parameters>(
+        VALUES (:account, :ufvk, :birthday_height, :recover_until_height)",
+        named_params![
+            ":account": u32::from(account),
+            ":ufvk": &key.encode(params),
+            ":birthday_height": u32::from(birthday.height()),
+            ":recover_until_height": birthday.recover_until().map(u32::from)
+        ],
     )?;
 
+    // If a birthday frontier is available, insert it into the note commitment tree. If the
+    // birthday frontier is the empty frontier, we don't need to do anything.
+    if let Some(frontier) = birthday.sapling_frontier().value() {
+        debug!("Inserting frontier into ShardTree: {:?}", frontier);
+        let shard_store = SqliteShardStore::<
+            _,
+            zcash_primitives::sapling::Node,
+            SAPLING_SHARD_HEIGHT,
+        >::from_connection(conn, SAPLING_TABLES_PREFIX)?;
+        let mut shard_tree: ShardTree<
+            _,
+            { zcash_primitives::sapling::NOTE_COMMITMENT_TREE_DEPTH },
+            SAPLING_SHARD_HEIGHT,
+        > = ShardTree::new(shard_store, PRUNING_DEPTH.try_into().unwrap());
+        shard_tree.insert_frontier_nodes(
+            frontier.clone(),
+            Retention::Checkpoint {
+                // This subtraction is safe, because all leaves in the tree appear in blocks, and
+                // the invariant that birthday.height() always corresponds to the block for which
+                // `frontier` is the tree state at the start of the block. Together, this means
+                // there exists a prior block for which frontier is the tree state at the end of
+                // the block.
+                id: birthday.height() - 1,
+                is_marked: false,
+            },
+        )?;
+    }
+
+    let sapling_activation_height = params
+        .activation_height(NetworkUpgrade::Sapling)
+        .expect("Sapling activation height must be available.");
+
+    // Add the ignored range up to the birthday height.
+    if sapling_activation_height < birthday.height() {
+        let ignored_range = sapling_activation_height..birthday.height();
+
+        replace_queue_entries::<SqliteClientError>(
+            conn,
+            &ignored_range,
+            Some(ScanRange::from_parts(
+                ignored_range.clone(),
+                ScanPriority::Ignored,
+            ))
+            .into_iter(),
+            false,
+        )?;
+    };
+
+    // Rewrite the scan ranges starting from the birthday height so that we'll ensure we
+    // re-scan to find any notes that might belong to the newly added account.
+    if let Some(t) = chain_tip {
+        let rescan_range = birthday.height()..(t + 1);
+
+        replace_queue_entries::<SqliteClientError>(
+            conn,
+            &rescan_range,
+            Some(ScanRange::from_parts(
+                rescan_range.clone(),
+                ScanPriority::Historic,
+            ))
+            .into_iter(),
+            true,
```

Comment:
```suggestion
            true, // force rescan
```

Reply:
Fixed in https://github.com/zcash/librustzcash/pull/935